src/components/Auth/AuthForm.tsx

```diff
@@ -0,0 +1,163 @@
+    }
+  };
+
+  const registerUser = () => {
+    return createUser({ name, email })
+      .then(saveUser);
+  };
+
+  const handleSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setErrorMessage('');
+    setIsLoading(true);
+
+    try {
+      if (needToRegister) {
+        await registerUser();
+      } else {
+        await loadUser();
+      }
+    } catch (error) {
+      setErrorMessage('Something went wrtong');
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  return (
+    <form onSubmit={handleSubmit} className="box mt-5">
+      <h1 className="title is-3">
+        {needToRegister
+          ? 'You need to register'
+          : 'Log in to open todos'}
+      </h1>
+
+      <div className="field">
+        <label className="label" htmlFor="user-email">
+          Email
+        </label>
+
+        <div
+          className={classNames('control has-icons-left', {
+            'is-loading': isLoading,
+          })}
+        >
+          <input
+            type="email"
+            id="user-email"
+            className={classNames('input', {
+              'is-danger': !needToRegister && errorMessage,
+            })}
+            placeholder="Enter your email"
+            disabled={isLoading || needToRegister}
+            value={email}
+            required
+            onChange={e => setEmail(e.target.value)}
+          />
+
+          <span className="icon is-small is-left">
+            <i className="fas fa-envelope" />
+          </span>
+        </div>
+
+        {!needToRegister && errorMessage && (
+          <p className="help is-danger">{errorMessage}</p>
+        )}
+      </div>
+
+      {needToRegister && (
+        <div className="field">
+          <label className="label" htmlFor="user-name">
+            Your Name
+          </label>
+
+          <div
+            className={classNames('control has-icons-left', {
+              'is-loading': isLoading,
+            })}
+          >
+            <input
+              type="text"
+              id="user-name"
+              className={classNames('input', {
+                'is-danger': needToRegister && errorMessage,
+              })}
+              placeholder="Enter your name"
+              required
+              minLength={4}
+              disabled={isLoading}
+              value={name}
+              onChange={e => setName(e.target.value)}
+            />
+
+            <span className="icon is-small is-left">
+              <i className="fas fa-user" />
+            </span>
+          </div>
+
+          {needToRegister && errorMessage && (
+            <p className="help is-danger">{errorMessage}</p>
+          )}
+        </div>
+      )}
+
+      <div className="field">
+        <button
+          type="submit"
+          className={classNames('button is-primary', {
+            'is-loading': isLoading,
+          })}
+        >
+          {needToRegister
+            ? 'Register'
+            : 'Login'}
```

Comment:
You can write it in one line in this case
```suggestion
          {needToRegister ? 'Register' : 'Login'}
```

Reply:
В мене так і було, проте Polosanya попросив мене так написати, то тепер і не знаю як правильно!